pkg/resources/spark/service.go

```diff
@@ -54,11 +55,67 @@ func NewHeadlessService(sc *dcv1alpha1.SparkCluster) *corev1.Service {
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      HeadlessServiceName(sc.Name),
 			Namespace: sc.Namespace,
-			Labels:    MetadataLabelsWithComponent(sc, ComponentMaster),
+			Labels:    MetadataLabelsWithComponent(sc, ComponentWorker),
 		},
 		Spec: corev1.ServiceSpec{
 			ClusterIP: corev1.ClusterIPNone,
 			Selector:  SelectorLabels(sc),
+			Ports:     []corev1.ServicePort{},
+			// TODO enable these ports for Istio support
+			// {
+			//	Name:       "cluster",
+			//	Port:       sc.Spec.ClusterPort,
+			//	TargetPort: intstr.FromString("cluster"),
+			// },
+			// {
+			//	Name:       "tcp-master-webport",
+			//	Port:       sc.Spec.TCPMasterWebPort,
+			//	TargetPort: intstr.FromString("http"),
+			//	Protocol:   corev1.ProtocolTCP,
+			// }, {
+			//	Name:       "tcp-worker-webport",
+			//	Port:       sc.Spec.TCPWorkerWebPort,
+			//	TargetPort: intstr.FromString("http"),
+			//	Protocol:   corev1.ProtocolTCP,
+			// },
+			// },
```

Comment:
This is just something to come and address in a later PR?

Reply:
right! Just changes that I've left in to uncomment when I get to working with Istio.